pkg/validation/spec/info.go

```diff
@@ -89,6 +89,32 @@ func (e Extensions) GetObject(key string, out interface{}) error {
 	return nil
 }
 
+// sanitized either uses:
+// - the map that was passed as an argument
+// - or, if the map is nil, returns the extension map if all the extensions were valid
+// - or allocates a new map.
+// In other words, if you specify a nil map as a parameter, don't edit the returned map.
+func (e Extensions) sanitized(m map[string]any) Extensions {
```

Comment:
I feel like this signature is too convoluted, I should probably split that into `sanitized` and `sanitizedWithMap` ... But I don't have a strong enough opinion ...